internal/component/common/loki/positions/positions_test.go

```diff
@@ -37,6 +39,90 @@ func tempFilename(t *testing.T) string {
 	return name
 }
 
+func TestConversion(t *testing.T) {
+	tmpDir := t.TempDir()
+	legacy := filepath.Join(tmpDir, "legacy")
+	positionsPath := filepath.Join(tmpDir, "positions")
+	legacyPositions := LegacyFile{
+		Positions: make(map[string]string),
+	}
+	// Filename and byte offset
+	legacyPositions.Positions["/tmp/random.log"] = "17623"
+	buf, err := yaml.Marshal(legacyPositions)
+	require.NoError(t, err)
+	err = os.WriteFile(legacy, buf, 0644)
+	require.NoError(t, err)
+	ConvertLegacyPositionsFile(legacy, positionsPath, log.NewNopLogger())
+	ps, err := readPositionsFile(Config{
+		PositionsFile: positionsPath,
+	}, log.NewNopLogger())
+	require.NoError(t, err)
+	require.Len(t, ps, 1)
+	for k, v := range ps {
+		require.True(t, k.Path == "/tmp/random.log")
+		require.True(t, v == "17623")
+	}
+	// Ensure old file is deleted.
+	_, err = os.Stat(legacy)
+	require.True(t, os.IsNotExist(err))
+}
+
+func TestConversionWithNewFile(t *testing.T) {
```

Comment:
something is a little unobvious with the test names especially this one. brainstorming here, what about in order:
- `TestLegacyConversion`
- `TestLegacyConversionNoOp` or `TestLegacyConversionPreexisting`
- `TestLegacyConversionNoFile`

Reply:
Good call